### ISSUE TITLE
Preview link doesn't override BlogPost PublishDate filter blocks previews

### ISSUE THREAD

**Issue by camfindlay** [silverstripe/silverstripe-sharedraftcontent#70]
Have just come across this on an install of CWP recipe 1.7.0.

Steps to replicate:
1. Install default CWP recipe 1.7.0
2. Create a `Blog` and add a `BlogPost` but don't set a date or set a future publish date then "save" but not "publish and publish" (legit use case while you're drafting blog content!).
3. Copy the shared draft content link and switch to another browser/incognito
4. Try to preview the draft blog post - you'll get a white screen no content.
5. Go back to the blog post from before and switch the date to now or a past publish date, save but not save and publish (blog is still a draft).
6. Get the shared draft token link again, go to your other window and paste.
7. You'll get the blog post draft!

I think the filtering that is hiding the post until a now or past publish date passes is blocking shared draft content from being able to preivew draft blog posts with no or a future date set.

This pattern is very common, writing a post, setting s future date, getting someone to view the post for sign off/peer review via a preview so I would expect this to function in this case.

Other draft pages that are not `BlogPost` don't have this issue.

**Comment by camfindlay** [comment]
@robbieaverill :+1:

**Comment by robbieaverill** [comment]
Reproduced, thanks @camfindlay

**Comment by robbieaverill** [comment]
The problem is that BlogPost is returning false for canView:

```php
        /**
         * @var SS_Datetime $publishDate
         */
        $publishDate = $this->dbObject('PublishDate');
        if(!$publishDate->exists()) {
            return false;
        }

        return !$publishDate->InFuture();
```

BlogPost returns false if parent class canView methods return false, but doesn't return true in the way that SiteTree does.

Perhaps we need to look at another way to render this content, maybe in a similar way to how the CMS preview is rendered.

cc @tractorcow - mind adding your thoughts when you have a chance?

**Comment by tractorcow** [comment]
We could add a hook into BlogPage and perhaps an extension on shared draft content that ensured the canView() was true in this case.

**Comment by robbieaverill** [comment]
@tractorcow @micmania1 I've submitted what I think is the most reasonable patch to https://github.com/silverstripe/silverstripe-blog/pull/487 to fix this issue, would you mind reviewing please?

**Comment by tractorcow** [comment]
I had a look at the fix and there is a flaw, but I see an alternative fix. :)

https://github.com/silverstripe/silverstripe-blog/pull/487#discussion_r148681768

Just make the PublishDate check only work on live (non-sharedraft).

**Comment by robbieaverill** [comment]
Fixed in blog 2.4.9 - thanks for reporting @camfindlay
